__tests__/components/TransactionList/TransactionRow.spec.ts

```diff
@@ -0,0 +1,888 @@
+        });
+
+        test('hide message icon', () => {
+            // Arrange:
+            const wrapper = getTransactionRowWrapper({ currentProfile: testnetProfile }, { transaction: mockTransferTransaction() });
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            // Act:
+            const hasMessage = vm.hasMessage();
+
+            // Assert:
+            expect(hasMessage).toBe(false);
+            expect(wrapper.find('.extend-icon-holder.message').exists()).toBe(false);
+        });
+
+        test('returns encrypted message title when message type is EncryptedMessage', () => {
+            // Arrange:
+            const recipient = getTestAccount('remoteTestnetRecipient');
+
+            const wrapper = getTransactionRowWrapper(
+                { currentProfile: testnetProfile },
+                {
+                    transaction: mockTransferTransaction(
+                        [],
+                        EncryptedMessage.create('test', recipient.publicAccount, currentSigner.privateKey),
+                    ),
+                },
+            );
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            // Act + Assert:
+            expect(vm.messagePayload).toBe('Encrypted message');
+        });
+
+        test('returns message payload in truncate when message length over 40', () => {
+            // Arrange:
+            const wrapper = getTransactionRowWrapper(
+                { currentProfile: testnetProfile },
+                { transaction: mockTransferTransaction([], PlainMessage.create('0'.repeat(45))) },
+            );
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            // Act + Assert:
+            expect(vm.messagePayload).toBe('0000000000000000000000000000000000000000...');
+        });
+
+        test('display mosaic icon when mosaic excluded network mosaic', () => {
+            // Arrange:
+            const wrapper = getTransactionRowWrapper(
+                { currentProfile: testnetProfile },
+                {
+                    transaction: mockTransferTransaction([new Mosaic(new MosaicId('461A9811A1DC1FBB'), UInt64.fromUint(0))]),
+                },
+            );
+
+            const vm = wrapper.vm as TransactionRowTs;
+
+            jest.spyOn(vm, 'isIncomingTransaction').mockReturnValue(true);
+
+            // Act:
+            const amount = vm.getAmount();
+            const getAmountMosaicId = vm.getAmountMosaicId();
+
+            // Assert:
+            expect(amount).toBe(0);
+            expect(getAmountMosaicId).toBe(undefined);
+            expect(wrapper.find('.extend-icon-holder.mosaic-icon').exists()).toBe(true);
+        });
+
+        test('returns non network mosaic when transaction type is transfer transaction', () => {
+            // Arrange:
+            const mockMosaic = [
+                { idHex: networkMock.currency.mosaicIdHex, amount: 100 },
+                { idHex: '461A9811A1DC1FBB', amount: 100 },
```

Comment:
perhaps another test that includes network mosaic to see that it is properly filtered ?

Reply:
it already included the network mosaic.

`{ idHex: networkMock.currency.mosaicIdHex, amount: 100 },`